monai/transforms/utility/array.py

```diff
@@ -251,3 +252,44 @@ def __init__(self, delay_time=0.0):
     def __call__(self, img, delay_time=None):
         time.sleep(self.delay_time if delay_time is None else delay_time)
         return img
+
+
+class KeepLargestConnectedComponent(Transform):
+    """
+    Keeps only the largest connected component.
+
+    Value 0 will be treated as background.
+    """
+    def __init__(self, is_independent=True):
+        """
+        Args:
+            is_independent (bool): consider several labels as a whole or independent, default is `True`.
+                Example use case would be segment label 1 is liver and label 2 is liver tumor, in that case
+                you want this "is_independent" to be specified as False.
+        """
+        super().__init__()
+        self.is_independent = is_independent
+
+    def __call__(self, img, applied_labels):
+        """
+        Args:
+            img (ndarray): shape must be (num_channels, H, W[, D]).
+            applied_labels (list or tuple of int): label number list for applying the connected component on.
+                The label that is not in this list will remain unchanged.
+        """
+        output = np.array(img)
```

Comment:
You can edit on `img` directly as the img will be useless after this transform in the transform chain.